src/benchmarks/micro/corefx/System.Threading.Tasks.Dataflow/Perf.BroadcastBlock.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Threading.Tasks.Dataflow.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX)]
+    public class BroadcastBlockPerfTests : ReceivablePropagatorPerfTests<BroadcastBlock<int>, int>
+    {
+        public override BroadcastBlock<int> CreateBlock() =>
+            new BroadcastBlock<int>(
+                i => i,
+                new ExecutionDataflowBlockOptions
+                {
+                    MaxDegreeOfParallelism = Environment.ProcessorCount
+                });
+
+        [Benchmark(OperationsPerInvoke = MessagesCount)]
+        public Task PostMultiReceiveParallel() => MultiParallel(() => Post());
+
+        [Benchmark(OperationsPerInvoke = MessagesCount)]
+        public Task SendMultiReceiveAsyncParallel() => MultiParallel(() => SendAsync());
+
+        private async Task MultiParallel(Func<Task> doTask)
+        {
+            var options = new DataflowLinkOptions { PropagateCompletion = true };
+            var action1 = new ActionBlock<int>(i => { });
+            var action2 = new ActionBlock<int>(i => { });
+            block.LinkTo(action1, options);
+            block.LinkTo(action2, options);
+
+            await doTask();
+            block.Complete();
```

Comment:
It looks like CreateBlock is called from the GlobalSetup, which means it's only going to be invoked once per test, regardless of the number of times the benchmark method is invoked.  Which means after the first time, the block will already have been completed.

Reply:
Moved GlobalSetup to IterationSetup.
Does it solve the issue ?

Reply:
I believe it addresses the issue, but potentially creates another. @adamsitnik, this is going to result in insufficient iteration counts by default, right?

Reply:
@stephentoub is right. In such case we should most probably call `CreateBlock` from the benchmark itself.

Reply:
Fixed.